pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolHandler.java

```diff
@@ -0,0 +1,590 @@
+            // the mechanism declared by the Message-Encoding header. A value of 0 indicates that no encoding of
+            // Message bytes has occurred.
+            //
+            // This seems to support the notion that compression can be enabled or disabled irrespective of the
+            // grpc-accept-encoding header.
+
+            // FUTURE: If the client sends a "grpc-accept-encoding", and if we support one of them, then we should pick
+            // one and use it in the response. Otherwise, we should not have any compression.
+
+            // If the grpc-timeout header is present, determine when that timeout would occur, or default to a future
+            // that is so far in the future it will never happen.
+            final var timeout = requestHeaders.value(GRPC_TIMEOUT);
+            deadlineFuture = timeout.isPresent()
+                    ? scheduleDeadline(timeout.get())
+                    : new NoopScheduledFuture();
+
+            // At this point, the request itself is valid. Maybe it will still fail to be handled by the service
+            // interface, but as far as the protocol is concerned, this was a valid request. Send the headers
+            // back to the client (the messages and trailers will be sent later).
+            sendResponseHeaders(GRPC_ENCODING_IDENTITY, requestContentType, emptyList());
+
+            // NOTE: The specification mentions the "Message-Type" header. Like everyone else, we're just going to
+            // ignore that header. See https://github.com/grpc/grpc/issues/12468.
+
+            // FUTURE: Should we support custom metadata, we would extract it here and pass it along via "options".
+            // We should have a wrapper around them, such that we don't process the custom headers ourselves, but
+            // allow the service interface to look up special headers based on key.
+
+            // Create the "options" to make available to the ServiceInterface. These options are used to decide on the
+            // best way to parse or handle the request.
+            final var options = new Options(
+                    headers.authority(), // the client (see http2 spec)
+                    contentType.equals(APPLICATION_GRPC_PROTO),
+                    contentType.equals(APPLICATION_GRPC_JSON),
+                    contentType);
+
+            // Setup the subscribers. The "outgoing" subscriber will send messages to the client. This
+            // is given to the "open" method on the service to allow it to send messages to the client.
+            final var outgoing = new SendToClientSubscriber();
+            incoming = route.service().open(route.method(), options, outgoing);
+        } catch (final GrpcException grpcException) {
+            route.failedGrpcRequestCounter().increment();
+            new TrailerOnlyBuilder().grpcStatus(grpcException.status())
+                    .statusMessage(grpcException.getMessage())
+                    .send();
```

Comment:
I agree that the nested classes can be confusing.

For now, I think I'll leave it as-is, since Richard is wanting this code to merge as a baseline, and then be modified / improved in the future. Since I'm taking over a bit to try to get this merged, it will allow me to avoid doing a lot of rewrites of Richard's code